app/src/main/java/space/taran/arknavigator/mvp/model/repo/PlainResourcesIndex.kt

```diff
@@ -87,13 +102,12 @@ class PlainResourcesIndex internal constructor (
         }
 
         val toInsert = diff.updated + diff.added
-        toInsert.forEach {
-            val id = computeId(it)
-
-            metaByPath[it] = ResourceMeta(
-                id = id,
-                modified = Files.getLastModifiedTime(it))
-            pathById[id] = it
+        toInsert.forEach { path ->
+            val meta = ResourceMeta.fromPath(path)
+            metaByPath[path] = meta
+            pathById[meta.id] = path
+            if (isFormat(path, "pdf"))
```

Comment:
This logic should not be here, just call `previewsRepo.generatePreview(...)` and dispatch on type there. If no preview needed for certain type, then just do nothing inside `generatePreview`.

Reply:
Also, this generation is called twice in some cases, since the same call is placed in `persistResources`, which is called from `reindexRoot`.